src/main/Config.h

```diff
@@ -338,6 +338,10 @@ class Config : public std::enable_shared_from_this<Config>
     // fees, and scp history in the database
     bool MODE_STORES_HISTORY_MISC;
 
+    // A config parameter that stores a minimal amount of historical data in the
+    // database (i.e. no meta)
+    bool MODE_STORES_HISTORY_MINIMAL;
```

Comment:
I'm not sure we need a new mode here: the proposed approach in https://github.com/stellar/stellar-core/issues/4211 recommends re-using MODE_STORES_HISTORY_MISC in captive core mode to avoid populating history tables completely (note that if we do it, we should enforce that core isn't configured to publish)

Reply:
AIUI, the issue identified the two use cases for optimization, the captive core case (the requirements of which are met by the `MODE_STORES_HISTORY_MISC`)  and the validator use case (write history, but not meta)?

Reply:
cc @MonsieurNicolas

Reply:
Currently captive core still writes meta, because on typical deployments MODE_STORES_HISTORY_MISC is on by default (it's only off in in-memory mode). So to address both cases you can:
* have MODE_STORES_HISTORY_MISC off on all captive core configurations (need to verify that core isn't configured to publish history in this case and throw if it is)
* for validators, we can update storeTransaction to write empty meta (this is already the case in the PR), and not call `storeUpgradeHistory` and `storeTransactionFee`.
* For all other configurations, we need to continue writing tx meta, and probably add a deprecation warning on startup

Reply:
Btw, looks like for the first bullet we already ensure that publish should be disabled if history is not stored: https://github.com/stellar/stellar-core/blob/97138128b4fc02cf296a07acc5d7145557a27782/src/main/ApplicationImpl.cpp#L770

Reply:
> have MODE_STORES_HISTORY_MISC off on all captive core configurations (need to verify that core isn't configured to publish history in this case and throw if it is)

This is controlled by horizon / validator operators right? 

> for validators, we can update storeTransaction to write empty meta (this is already the case in the PR), and not call storeUpgradeHistory and storeTransactionFee.

I'm a little confused how this is exposed, these are recommendations for configuration rather than sets of configs we apply right? Ie. we would "recommend" captive core sets MODE_STORES_HISTORY_MISC = false, and that validators set `MODE_STORES_HISTORY_MINIMAL= true`. 

If that is the case, It could be helpful to provide an opinionated "default validator options" etc as there are a lot of knobs to turn.

Reply:
Yeah thanks that's makes much more sense.  I was thinking it would take some attention to detail for operators to get this right so I'm glad we have higher level configs that result in these being set.